### PR TITLE
refactor recoGenRecommander

### PR DESCRIPTION
**Pull request by lawzlo**
separate recoGen and publicFeedGen to make classes smaller and easy to maintain.

@stkem @andrewconner @yingjieMiao

### CODE REVIEW

**Comment by lawzlo** [comment]
tested, pushing to prod.

File: kifi-backend/modules/curator/app/com/keepit/curator/commanders/PublicFeedGenerationCommander.scala [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    db.readOnlyReplicaAsync { implicit session =>
+      publicFeedRepo.getByTopMasterScore(howManyMax)
+    }
+  }
+
+  private def computePublicMasterScore(scores: PublicUriScores): Float = {
+    (1 * scores.recencyScore +
+      1 * scores.popularityScore +
+      6 * scores.rekeepScore +
+      3 * scores.discoveryScore) *
```
Comment (stkem) [inline kifi-backend/modules/curator/app/com/keepit/curator/commanders/PublicFeedGenerationCommander.scala:47]: Why did you remove the curation score?

Comment (lawzlo) [inline kifi-backend/modules/curator/app/com/keepit/curator/commanders/PublicFeedGenerationCommander.scala:47]: oops. missed during refactoring. adding back.

Comment (stkem) [inline kifi-backend/modules/curator/app/com/keepit/curator/commanders/PublicFeedGenerationCommander.scala:47]: Please be more careful with this. You shouldn't loose logic when moving a function from one class to another.

Comment (lawzlo) [inline kifi-backend/modules/curator/app/com/keepit/curator/commanders/PublicFeedGenerationCommander.scala:47]: definitely. I will check again.

File: kifi-backend/modules/curator/app/com/keepit/curator/commanders/PublicFeedGenerationCommander.scala [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  PublicFeedRepo,
+  PublicSeedItem
+}
+import com.keepit.model.{ ExperimentType, User, Name, SystemValueRepo }
+import com.keepit.shoebox.ShoeboxServiceClient
+
+import scala.concurrent.Future
+import play.api.libs.concurrent.Execution.Implicits.defaultContext
+
+class PublicFeedGenerationCommander @Inject() (
```
Comment (stkem) [inline kifi-backend/modules/curator/app/com/keepit/curator/commanders/PublicFeedGenerationCommander.scala:22]: This needs to be a Singleton, otherwise the `pubicFeedsGenerationLock` does nothing, since there will be a new one created every time this class is injected somewhere.

Comment (lawzlo) [inline kifi-backend/modules/curator/app/com/keepit/curator/commanders/PublicFeedGenerationCommander.scala:22]: make sense. Adding it. Thanks.

File: kifi-backend/modules/curator/app/com/keepit/curator/commanders/PublicFeedGenerationCommander.scala [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    publicFeedRepo: PublicFeedRepo,
+    systemValueRepo: SystemValueRepo,
+    experimentCommander: RemoteUserExperimentCommander) {
+
+  val pubicFeedsGenerationLock = new ReactiveLock(1)
+  private val SEQ_NUM_NAME: Name[SequenceNumber[PublicSeedItem]] = Name("public_feeds_seq_num")
+  private def specialCurators(): Future[Seq[Id[User]]] =
+    experimentCommander.getUsersByExperiment(ExperimentType.SPECIAL_CURATOR).map(users => users.map(_.id.get).toSeq)
+
+  def getPublicFeeds(howManyMax: Int): Future[Seq[PublicFeed]] = {
```
Comment (lawzlo) [inline kifi-backend/modules/curator/app/com/keepit/curator/commanders/PublicFeedGenerationCommander.scala:37]: I found this API never be used. Instead, recoRetrievalCommander calls feedRepo directly to get feeds. Is there anywhere else may use it? Or I can just remove it.

Comment (stkem) [inline kifi-backend/modules/curator/app/com/keepit/curator/commanders/PublicFeedGenerationCommander.scala:37]: I think you can remove it

Comment (lawzlo) [inline kifi-backend/modules/curator/app/com/keepit/curator/commanders/PublicFeedGenerationCommander.scala:37]: :+1:
